### ISSUE TITLE
3.1.31 regression: -sMODULARIZE and -sSTRICT_JS don't work together 

### ISSUE THREAD

**Issue by hoodmane** [emscripten-core/emscripten#18620]
Minimal reproduction:
```sh
echo "int main(){}" > tmp.c
emcc tmp.c -o ./tmp.js -sSTRICT_JS -sMODULARIZE
node tmp.js
```
With Emscripten 3.1.30 this outputs nothing and returns with exit status 0. With Emscripten 3.1.31 it throws:
```
SyntaxError: Illegal 'use strict' directive in function with non-simple parameter list
```

**Comment by hoodmane** [comment]
The difference is caused by:
```
  return function (config) {
    var Module = config || {};
    "use strict";
```
got changed to
```
  return function (Module = {}) {
    // include: shell.js
    "use strict";
```

**Comment by sbc100** [comment]
I believe this was fixed already in #18614 ?

**Comment by hoodmane** [comment]
Looks like. But we aren't building an ES6 module, but we do want `-sMODULARIZE`, so `-sSTRICT_JS` is actually useful to us. But I guess we can just inject "use strict" at the top of the file with sed like we've been doing.

MDN suggests this as workaround:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Strict_Non_Simple_Params#function_statement

**Comment by sbc100** [comment]
I see, I guess we could inject `use strict` outside of the modularization closure maybe?  Would that work?

**Comment by hoodmane** [comment]
Yeah I think the correct thing would be to just put it at the top of the file.
